### ISSUE TITLE
fatal error: 'folly/dynamic.h' file not found

### ISSUE THREAD

**Issue by only1chi** [software-mansion/react-native-gesture-handler#2275]
### Description

I'm trying to build an android project with a bare app (in a monorepo), with React Native 0.70.1. 
There is a similar issue (now closed) for a RN 68.1, but the solution did not work for me. I have cleaned my project and cleared the cache. 

I was building with:
- `react-native-reanimated v3.0.0-rc3`
- `react-native-safe-context-area v4.4.1`

I am getting the following build error:

```
FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':react-native-gesture-handler:buildCMakeDebug[arm64-v8a]'.
> Build command failed.
  Error while executing process /Users/chizurokechikwendu/Library/Android/sdk/cmake/3.18.1/bin/ninja with arguments {-C /Users/chizurokechikwendu/Desktop/Git_Repo/janus/packages/testapp/node_modules/react-native-gesture-handler/android/.cxx/Debug/4v5z244u/arm64-v8a gesturehandler}
  ninja: Entering directory `/Users/chizurokechikwendu/Desktop/Git_Repo/janus/packages/testapp/node_modules/react-native-gesture-handler/android/.cxx/Debug/4v5z244u/arm64-v8a'
  [1/2] Building CXX object CMakeFiles/gesturehandler.dir/cpp-adapter.cpp.o
  FAILED: CMakeFiles/gesturehandler.dir/cpp-adapter.cpp.o 
  /Users/chizurokechikwendu/Library/Android/sdk/ndk/24.0.8215888/toolchains/llvm/prebuilt/darwin-x86_64/bin/clang++ --target=aarch64-none-linux-android26 --sysroot=/Users/chizurokechikwendu/Library/Android/sdk/ndk/24.0.8215888/toolchains/llvm/prebuilt/darwin-x86_64/sysroot -Dgesturehandler_EXPORTS -I/Users/chizurokechikwendu/Desktop/Git_Repo/janus/packages/testapp/node_modules/react-native/ReactAndroid/build/third-party-ndk/boost/boost_1_76_0 -I/Users/chizurokechikwendu/Desktop/Git_Repo/janus/packages/testapp/node_modules/react-native/ReactAndroid/build/third-party-ndk/double-conversion -I/Users/chizurokechikwendu/Desktop/Git_Repo/janus/packages/testapp/node_modules/react-native/ReactAndroid/build/third-party-ndk/folly -isystem /Users/chizurokechikwendu/Desktop/Git_Repo/janus/packages/testapp/node_modules/react-native/ReactCommon/jsi -isystem /Users/chizurokechikwendu/Desktop/Git_Repo/janus/packages/testapp/node_modules/react-native/ReactCommon/react/renderer/uimanager -isystem /Users/chizurokechikwendu/Desktop/Git_Repo/janus/packages/testapp/node_modules/react-native/ReactCommon -isystem /Users/chizurokechikwendu/Desktop/Git_Repo/janus/packages/testapp/node_modules/react-native/ReactCommon/react/renderer/core -DFOLLY_NO_CONFIG=1 -DFOLLY_HAVE_CLOCK_GETTIME=1 -DFOLLY_USE_LIBCPP=1 -DFOLLY_MOBILE=1 -DFOLLY_HAVE_RECVMMSG=1 -DFOLLY_HAVE_PTHREAD=1 -fno-limit-debug-info  -fPIC -std=gnu++17 -MD -MT CMakeFiles/gesturehandler.dir/cpp-adapter.cpp.o -MF CMakeFiles/gesturehandler.dir/cpp-adapter.cpp.o.d -o CMakeFiles/gesturehandler.dir/cpp-adapter.cpp.o -c /Users/chizurokechikwendu/Desktop/Git_Repo/janus/packages/testapp/node_modules/react-native-gesture-handler/android/src/main/jni/cpp-adapter.cpp
  In file included from /Users/chizurokechikwendu/Desktop/Git_Repo/janus/packages/testapp/node_modules/react-native-gesture-handler/android/src/main/jni/cpp-adapter.cpp:4:
  /Users/chizurokechikwendu/Desktop/Git_Repo/janus/packages/testapp/node_modules/react-native/ReactCommon/react/renderer/uimanager/primitives.h:10:10: fatal error: 'folly/dynamic.h' file not found
  #include <folly/dynamic.h>
           ^~~~~~~~~~~~~~~~~
  1 error generated.
  ninja: build stopped: subcommand failed.
```

### Steps to reproduce

1. Configure project to use React Native 0.70.1, and enable new architecture
2. run yarn start --reset-cachce
3. run yarn workspaces @myworkspace/myapp react-native run-android

### Snack or a link to a repository

Private Repo

### Gesture Handler version

2.7.1

### React Native version

0.70.1

### Platforms

Android

### JavaScript runtime

Hermes

### Workflow

React Native (without Expo)

### Architecture

Fabric (New Architecture)

### Build type

Debug mode

### Device

Android emulator

### Device model

N/A

### Acknowledgements

Yes

**Comment by j-piasecki** [comment]
I tried to set up a new app with the same dependencies and it worked for me, that said could you:
- try building your app without Gesture Handler to make sure it's the culprit
- if it builds without GH, are you able to give more information about your setup, like whether `react-native` and `react-native-gesture-handler` are hoisted or not?

**Comment by only1chi** [comment]
I did try to build my app without Gesture Handler, but I think that gradle cache is not properly cleaned. I did run the command `.\gradlew clean` but I get the following error:

```
chizurokechikwendu@chizs-m1-mac android % ./gradlew clean            
Configuration on demand is an incubating feature.
:ReactNative:Unexpected empty result of running '[node, -e, try {console.log(require('@react-native-community/cli').bin);} catch (e) {console.log(require('react-native/cli').bin);}]' command.
:ReactNative:Running '[node, -e, try {console.log(require('@react-native-community/cli').bin);} catch (e) {console.log(require('react-native/cli').bin);}]' command failed.

FAILURE: Build failed with an exception.

* Where:
Script '/Users/chizurokechikwendu/Desktop/Git_Repo/janus/node_modules/@react-native-community/cli-platform-android/native_modules.gradle' line: 413

* What went wrong:
A problem occurred evaluating script.
> /Users/chizurokechikwendu/Desktop/Git_Repo/janus/node_modules/@react-native-community/cli/build/index.js:156    cmd.option(opt.name, opt.description ?? '', opt.parse || (val => val), typeof opt.default === 'function' ? opt.default(rest[0]) : opt.default);                                          ^SyntaxError: Unexpected token '?'    at wrapSafe (internal/modules/cjs/loader.js:915:16)    at Module._compile (internal/modules/cjs/loader.js:963:27)    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)    at Module.load (internal/modules/cjs/loader.js:863:32)    at Function.Module._load (internal/modules/cjs/loader.js:708:14)    at Module.require (internal/modules/cjs/loader.js:887:19)    at require (internal/modules/cjs/helpers.js:74:18)    at Object.<anonymous> (/Users/chizurokechikwendu/Desktop/Git_Repo/janus/node_modules/react-native/cli.js:13:11)    at Module._compile (internal/modules/cjs/loader.js:999:30)    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 531ms
```

I know it has to do with my monorepo setup.

- I have tried to run this command WITHOUT node modules hoisted and I was unsuccessful.
- I have tried to run it WITH node modules hoisted, and I was also unsuccessful.

The package.json in myworkspace/myapp is has this option:

```
  "installConfig": {
    "hoistingLimits": "none"
  }
```

I've tried combinations where I've set hoistingLimits to `workspaces`, `dependencies` and `none`. 
None of them have worked.

**Comment by only1chi** [comment]
@j-piasecki 
I eventually got things to build, although I had to give up some of the goals that I had with the monorepo.
So I setup the package/app with the option of hoisting node modules in `workspaces`. This essentially reproduced the node-modules in the package/app sub folder.
Then I configured the `build.gradle` and `app/build.gradle` with typical relative paths to node-modules. Once I did that I could build the package. 

It appears that `react-native-gesture-handler` needs to be in the `..\node_modules\...` folder location in order for it to build properly.
